app/helpers/service_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ServiceHelper
+  private
+
+  def custom_attr_fields(host, custom_attrs)
+    validated_fields = []
+    custom_attrs.each do |attr_name, fields|
+      fields = downcase_keys(fields)
+      value = fields['value'].to_s
+      fields['value'] = filter_vulnerabilities(host, attr_name, value)
+      validated_fields << fields
+    end
+    validated_fields
+  end
+
+  def downcase_keys(raw_hash)
+    hash_with_downcase_key = {}
+    raw_hash.each_pair { |k, v| hash_with_downcase_key[k.downcase] = v }
+    hash_with_downcase_key
+  end
+
+  # rubocop:disable Security/Eval, Style/StringLiterals, Lint/UnusedMethodArgument
+  def filter_vulnerabilities(host, attr_name, value)
+    return value unless value.include?('#{')
+    error_message = "custom attribute '#{attr_name}' configuration contains insecure interpolation"
+    raise error_message unless contains_secure_host_attr?(value)
+    eval("\"" + value + "\"")
+  rescue StandardError => e
+    raise e.message
+  end
+  # rubocop:enable Security/Eval, Style/StringLiterals, Lint/UnusedMethodArgument
+
+  def contains_secure_host_attr?(value)
```

Comment:
We don't want this extra checking. I want the ability to put ANY code in the string from the config file. Again the person making the config file can be trusted.

Reply:
understood